core/domain/suggestion_registry.py

```diff
@@ -0,0 +1,329 @@
+# Copyright 2018 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Registry for Oppia suggestions. Contains a BaseSuggestion class and
```

Comment:
Though you can call it a registry but doesn't strictly looks/works like a registry used in Oppia (for e.g. the interaction registry).

Reply:
Discussed it on chat. Decided we'll go ahead with this for now.